nautobot/ui/src/components/ObjectListTable.js

```diff
@@ -158,14 +157,19 @@ export default function ObjectListTable({
                             Actions
                         </UIButton>
                         <Icon.TbMinusVertical />
-
-                        <RouterButton
+                        <UIButton
                             to={`${location.pathname}add/`}
                             size="sm"
                             leftIcon={<PlusIcon />}
+                            onClick={(e) => {
+                                e.preventDefault();
+                                if (process.env.NODE_ENV === "production") {
```

Comment:
What's the purpose of / need for this check? Could you maybe add some explanatory comments?